### PR TITLE
Affirm Previous Report Incorporation

### LINKED ISSUE #41: incorporate comments from previous reports
Investigate ability to carry forward comments from previous excel outputs into new excel outputs.

This is especially useful when rows are flagged for review but are deemed ok as-is by data managment (no need to change programming for the flag).

### PR DESCRIPTION
**Pull request by Meghansaha**
Closes #41

### CODE REVIEW

**Comment by Meghansaha** [comment]
# `openxlsx2` Issue 
## Details
The checks (Github Actions) for a [previous commit](https://github.com/pcctc/affirm/pull/44/checks?check_run_id=37743048991) failed unexpectedly despite being fine in earlier commits on this PR. The failures came after I updated the `openxlsx` package from version [1.12](https://janmarvin.github.io/openxlsx2/news/index.html#openxlsx2-112) to [1.13](https://janmarvin.github.io/openxlsx2/news/index.html#openxlsx2-113).

The issue seems to be with the `wb_add_font()`'s `bold` and `italic` arguments. This function is within `affirm`'s internal [`utils-report.R`](https://github.com/pcctc/affirm/blob/main/R/utils-report.R) script to assist in creating the `affirm_report_excel()` output.

In our last commit that was rolled into main _(using `openxlsx2` < 1.13)_ these code chunks ran fine:

https://github.com/pcctc/affirm/blob/6fe15ddb3b8d5162218eecfc5c33756d495668d3/R/utils-report.R#L220-L224

https://github.com/pcctc/affirm/blob/6fe15ddb3b8d5162218eecfc5c33756d495668d3/R/utils-report.R#L241-L247

But if this code is now ran with `openxlsx2` version 1.13, an error is now thrown:

```
Error: b must be "0" or "1"
```
> [!important]
>Given the error, I took a guess and assumed it wanted a Boolean `TRUE`/`FALSE` for these arguments. In my testing, passing `TRUE`/`FALSE` to the `bold` and `italic` arguments did get the error to stop, and it **did still produce the expected styling** in the excel output.

### Proposed Resolutions:
> [!note]
> I think we can handle in different ways:
> 1. Restrict version of `openxslx` <= 2.12 in `affirm`'s `DESCRIPTION` file for now and use originally accepted values in `wb_add_font()`
> 2. Replace the original values of `wb_add_font()` with proposed Booleans as it does appear to work in all reprexes below (versions 1.13, 1.12, and the minimum 1.1 specified in the affirm `DESCRIPTION` file)
>
> Either way, [filing an issue](https://github.com/JanMarvin/openxlsx2/issues) for the `openxlsx2` package may be warranted.

<br>

----

## General Notes
+ wb_add_font() is [covered](https://app.codecov.io/gh/JanMarvin/openxlsx2/blob/main/R%2Fclass-workbook-wrappers.R#L3427) for testing within the openxlsx2 package, but current tests appear to use different inputs for the `bold` and `italic` arguments. An example is [here](https://github.com/JanMarvin/openxlsx2/blob/dd8d617fdda38c8317254986f4b91700e7db245f/tests/testthat/test-wb_styles.R#L520-L533)
    - There doesn't appear to be any current instances where the values "double" or "italic" is tested currently for this function.

+ In the [example used in `openxlsx2`'s help file](https://janmarvin.github.io/openxlsx2/reference/wb_add_font.html?q=wb_add_font#ref-examples), the `bold` and `italic` arguments are not used. So no error was caught here, allowing the package to be compiled:

```
wb <- wb_workbook() %>% wb_add_worksheet("S1") %>% wb_add_data("S1", mtcars)
#wb %>% wb_add_font("S1", "A1:K1", name = "Arial", color = wb_color(theme = "4"))
```
+ There doesn't appear to be any relevant breaking changes listed within `openxlsx2`'s changelogs directly related to `wb_add_font()`
    - Version 1.13 [Changelog](https://janmarvin.github.io/openxlsx2/news/index.html#openxlsx2-113): (CRAN Release: 2025-02-15)
    - Version 1.12 [Changelog](https://janmarvin.github.io/openxlsx2/news/index.html#openxlsx2-112): (CRAN Release: 2024-12-19)

+ There's only two commits between these `openxlsx2` versions that modify the R script `class-workbook-wrappers.R` where `wb_add_font()` lives:
   - Commit [6024d0f](https://github.com/JanMarvin/openxlsx2/commit/6024d0f3b148d64d950d18a8ebd4f1806d89535d#diff-635bc1ddefececfcea5d2996c31fc8b97496c17abd1c5b6cf881b9ea8ce0b8fa) on 2025-01-04
   - Commit [2e5fd1d](https://github.com/JanMarvin/openxlsx2/commit/2e5fd1de603d4e252eb63c35823227b7ac88b000#diff-635bc1ddefececfcea5d2996c31fc8b97496c17abd1c5b6cf881b9ea8ce0b8fa) on 2025-01-04
       - None of these changes directly touch the `wb_add_font()` fx 🤷🏾‍♀️

<br>

---
## Reprexes
### Version 1.12
```
# Optional rstudioapi for session restart
#install.packages("rstudioapi)

# Get current version of the openxlsx pkg on the machine
current_vers <- packageVersion("openxlsx2")

#==============================================================================#
# Version 1.12------------------------------------------------------------------
#==============================================================================#
# Version 1.12 is what I started building on....
remotes::install_version("openxlsx2", version = "1.12")
library(openxlsx2)

# Init workbook
wb <- wb_workbook() |> wb_add_worksheet("S1") |> wb_add_data("S1", mtcars)

# The original values for "bold" and "italic":
wb |>
  wb_add_font(
    "S1", "A1:K1",
    name = "Arial",
    bold = "double",
    italic = "italic"
    )

# Make temp file
tempxlsx_v1.12 <- tempfile(pattern = "_v1.12_", fileext = ".xlsx")

# Save
wb_save(
  wb,
  file = tempxlsx_v1.12
)

# Open the file to investigate styling, styles as intended
browseURL(tempxlsx_v1.12)

# Clean and Restart the session
detach("package:openxlsx2", unload = TRUE)
rstudioapi::restartSession(clean = TRUE)

```
### Version 1.13

```
#==============================================================================#
# Version 1.13------------------------------------------------------------------
#==============================================================================#
# After work was done, I updated to 1.13 as per package dev work checklist...
remotes::install_version("openxlsx2", version = "1.13")
library(openxlsx2)

wb <- wb_workbook() |> wb_add_worksheet("S1") |> wb_add_data("S1", mtcars)

# This don't work (╯°□°)╯︵ ┻━┻
# Throws error: (Error: b must be "0" or "1")
tryCatch(
  {
    wb |>
    openxlsx2::wb_add_font(
      "S1", "A1:K1",
      name = "Arial",
      bold = "double",
      italic = "italic"
    )
  },
  error = function(e) {
    message("Note: The following error occurred but was caught: ", e$message)
  }
)
  
# But this does work...
wb |>
  wb_add_font(
    "S1", "A1:K1",
    name = "Arial",
    bold = TRUE,
    italic = TRUE,
  )

# Make temp file
tempxlsx_v1.13 <- tempfile(pattern = "_v1.13_", fileext = ".xlsx")

#Save
wb_save(
  wb,
  file = tempxlsx_v1.13
)

# Open the file to investigate styling, styles as intended
browseURL(tempxlsx_v1.13)

# Clean and Restart the session
detach("package:openxlsx2", unload = TRUE)
rstudioapi::restartSession(clean = TRUE)

```
### Version 1.12 Recheck With Booleans
```
#==============================================================================#
# Version 1.12 Recheck With Booleans--------------------------------------------
#==============================================================================#
remotes::install_version("openxlsx2", version = "1.12")
library(openxlsx2)

# Init workbook
wb <- wb_workbook() |> wb_add_worksheet("S1") |> wb_add_data("S1", mtcars)

# Replacing the original values for "bold" and "italic" with `TRUE` works:
wb |>
  wb_add_font(
    "S1", "A1:K1",
    name = "Arial",
    bold = TRUE,
    italic = TRUE
  )

# Make temp file
tempxlsx_v1.12_bool <- tempfile(pattern = "_v1.12_bool_", fileext = ".xlsx")

# Save
wb_save(
  wb,
  file = tempxlsx_v1.12_bool
)

# Open the file to investigate styling, styles as intended
browseURL(tempxlsx_v1.12_bool)

# Clean and Restart the session
detach("package:openxlsx2", unload = TRUE)
rstudioapi::restartSession(clean = TRUE)

```

### Version 1.1 Check With Booleans
```
#==============================================================================#
# Version 1.1 Check With Booleans---------------------------------------------
#==============================================================================#
remotes::install_version("openxlsx2", version = "1.1")
library(openxlsx2)

# Init workbook
wb <- wb_workbook() |> wb_add_worksheet("S1") |> wb_add_data("S1", mtcars)

# The original values for "bold" and "italic":
wb |>
  wb_add_font(
    "S1", "A1:K1",
    name = "Arial",
    bold = TRUE,
    italic = TRUE
  )

# Make temp file
tempxlsx_v1.1_bool <- tempfile(pattern = "_v1.1_bool_", fileext = ".xlsx")

# Save
wb_save(
  wb,
  file = tempxlsx_v1.1_bool
)

# Open the file to investigate styling, styles as intended
browseURL(tempxlsx_v1.1_bool)

# Clean and Restart the session
detach("package:openxlsx2", unload = TRUE)
rstudioapi::restartSession(clean = TRUE)

# install original version back onto machine
remotes::install_version("openxlsx2", version = current_vers)
```

**Comment by JanMarvin** [comment]
Hi @Meghansaha , just came across this issue. The error message from `openxlsx2` is unfortunately not intuitive, but at least it correctly identified an issue in your code. The code above should be 
```R
wb_add_font(..., bold = TRUE, italic = TRUE)
```
and not 
```R
wb_add_font(..., bold = "bold", italic = "italic")
```
If it worked previously, it probably was a fluke and it is not guaranteed to not cause issues with certain spreadsheet software.

The underlying XML code accepts only a boolean argument and `"bold"`, `"italic"`, or `"double"` (which would be possible for underline) are not. It unfortunately took us a while to finally check these inputs.

**Comment by Meghansaha** [comment]
Hi @JanMarvin. Thank you so much for taking the time out to respond! I was hesitant about making an actual issue because I couldn't find any official documentation accepting `"bold"` and `"italic"` as proper inputs for those arguments. I inherited old code that included these values and just wasn't sure. Thank you for confirming this for me! I appreciate it.

**Comment by JanMarvin** [comment]
No problem! If it's not a bug report, you can use the `openxlsx2` discussion area to ask questions. Or if incorrectly opened as an issue, I will probably convert it to a discussion :-)

Below is an explanation of why the error message appears as it does. If you have a suggestion for a better message, please let me know—I wrote the current one, hoping no one would ever see it.  

Due to historical reasons, the underlying style creating functions, such as `create_font()`, accept a variety of arguments. By default, all these functions use `character` elements as an argument—this was an early design choice before I knew better. 

``` r
library(openxlsx2)
create_font(b = "1", color = "", family = "", name = "", scheme = "", sz = "")
#> [1] "<font><b val=\"1\"/></font>"
create_font(b = "true", color = "", family = "", name = "", scheme = "", sz = "")
#> [1] "<font><b val=\"true\"/></font>"

create_font(b = 1, color = "", family = "", name = "", scheme = "", sz = "")
#> [1] "<font><b val=\"1\"/></font>"
create_font(b = TRUE, color = "", family = "", name = "", scheme = "", sz = "")
#> [1] "<font><b val=\"1\"/></font>"
```

For `openxlsx2 1.12` passing `"bold"` would produce the following:
``` r
library(openxlsx2)
create_font(b = "bold", color = "", family = "", name = "", scheme = "", sz = "")
#> [1] "<font><b val=\"bold\"/></font>"
```

This does not cause issues in my MS365 Excel, but isn't exactly what the standard requests. The XML snippets are from the ECMA OOXML standard:
```XML
<!-- part of CT_Font -->
<xsd:element name="i" type="CT_BooleanProperty" minOccurs="0" maxOccurs="1"/>
<xsd:element name="b" type="CT_BooleanProperty" minOccurs="0" maxOccurs="1"/>
```
with
```XML
<xsd:complexType name="CT_BooleanProperty">
  <xsd:attribute name="val" type="xsd:boolean" use="optional" default="true"/>
</xsd:complexType>
```
And `xsd:boolean` accepts `0`, `1`, `true`, and `false`.  Unfortunately, with the many different spreadsheet applications available, it's difficult to determine which software supports which parts of the standard. For example, Excel is often the most strict—compared to LibreOffice and Numbers—likely because OOXML is a Microsoft standard, whereas other applications only implement parts of it and try not to fail on unsupported features. However, in this case, Excel doesn’t seem to enforce strict validation. That said, other software might, and future versions of Excel could as well. So, I decided to include the check.

**Comment by Meghansaha** [comment]
Thank you @JanMarvin. Going to take the time to digest all this when I have the bandwidth. Also, thank you for reminding me that Discussions exist, because I legitimately forgot 😅.

**Comment by Meghansaha** [comment]
>[!IMPORTANT]
>This works in development, but not production. I'm suspecting this is because:


1. **Too much time has passed with too many revisions to the affirmations in production**. I suggest that the work/testing for this is paused until we have a more recent 'previous' report to use as a more recent one than `2024-12-01` could not be provided. 

2. There may be some **additional logic that needs to be introduced to the missing column chunk of work**, As errors were thrown for newly created affirmations (when they should not have been):
https://github.com/pcctc/affirm/blob/40c3c3b7fc86812882ec3f59341ed3895c9074cb/R/utils-report.R#L414-L476

Based on the error output here, The affirmations (`AE011`,  `AE803`,  `ALL001`,  `ALL002`, `ALL003`, `ALL007`, `LAB012`) are newly created, so should not be here for mismatch checking:
```
✖ The current affirm report could not be updated due to missing columns in the following
  current affirmations:

1. Affirmation: `AE001`
  • `ae_record_position`
2. Affirmation: `AE011`
  •
3. Affirmation: `AE803`
  •
4. Affirmation: `ALL001`
  •
5. Affirmation: `ALL002`
  •
6. Affirmation: `ALL003`
  •
7. Affirmation: `ALL007`
  •
8. Affirmation: `ALL008`
  • `record_position`, `aeendat_int`, `aeacnoth`, `aeout`, `cmaeno`, `cmtrt`, `cmindc`,
  `cmstdat_int`, `cmendat_int`, and `flag_ae_cm1`
9. Affirmation: `ALL009`
  • `data_page_name`, `check`, `dupe_count`, `record_positions`, and `flag_duplicate`
10. Affirmation: `ALL010`
  • `check`, `record_positions`, and `flag_cross_duplicate`
11. Affirmation: `BS002`
  • `instance_name`, `visdat`, `visdat_raw`, `dddat`, and `flag_death`
12. Affirmation: `CM002`
  • `data_page_name`, `record_position`, `other`, `coded_val`, `field`, `pre_text`, and
  `other_fl`
13. Affirmation: `EOTEOS002`
  • `data_page_name`, `date_raw`, `record_position`, `field`, and `unkday_fl`
14. Affirmation: `EOTEOS003`
  • `date_raw` and `unkyear_fl`
15. Affirmation: `EX002`
  • `randomized_at_int`, `cmstdat_int`, `cmtrt`, `cmindc`, `cm_record_position`,
  `mhstdat_int`, `mhendat_int`, `mhterm`, `mh_record_position`, and `flag_mhterm`
16. Affirmation: `EX003`
  • `dsdecod_tx`, `dsstdat_tx_int`, `offtx_dsaeno`, and `flag_ae_fatal`
17. Affirmation: `EX004`
  • `dsdecod_st`, `dsdecod_tx`, `dsstdat_tx_int`, `offtx_dsaeno`, `dsscat_tx`, `exstdat_last`,
  `aemodify`, `aemodify_soc`, `aestdat_int`, and `flag_drugwithdrawn`
18. Affirmation: `EX005`
  • `dsdecod_tx`, `dsstdat_tx_int`, `exstdat_last`, and `flag_dsstdat_tx`
19. Affirmation: `LAB012`
  •
20. Affirmation: `LAB013`
  • `ic_dsstdat_int`, `tx`, `exstdat_int`, `expdose`, `exdstxt`, `exendat_int`, `exmod`,
  `exadj`, `flag_pred_overlap`, and `flag_first_date`
21. Affirmation: `LAB016`
  • `tx`, `exmod`, `exadj`, `exco`, `exstdat_int`, and `flag_tx_wo_ae`
22. Affirmation: `MH001`
  •
23. Affirmation: `MH002`
  •
24. Affirmation: `MH007`
  • `randomized_at`, `expdose`, `exdstxt`, `instance_name`, `visdat_int`, and
  `flag_exstdat_int`
25. Affirmation: `MH009`
  • `analyte_name`, `lab_flag`, `clin_sig_value`, `record_date`, `expected_aemodify`, `in_ae`,
  `aemodify`, `ae_match`, and `flag_ae`
26. Affirmation: `MH010`
  • `randomized_at`, `treatment_category`, `ADT_prior`, `mh_prdosu`, `cmgrpid`, `setmcrpc`,
  `cmongo`, `cmstdat_int`, `mh_cmendat_int`, `prohibited_therapy`,
  `prohibited_after_randomization`, `prohibited_ongoing`, `started_tx`, `mhsg_prtrt`,
  `orch_prior`, and `flag_ADT_prior`
27. Affirmation: `MR801`
  • `treatment_category`, `ADT_prior`, `mh_prdosu`, `cmgrpid`, `setmcrpc`, `cmongo`,
  `cmstdat_int`, `mh_cmendat_int`, `prohibited_therapy`, `prohibited_after_randomization`,
  `prohibited_ongoing`, `started_tx`, `mhsg_prtrt`, `orch_prior`, and `flag_prohibited_prior`
28. Affirmation: `MR802`
  • `randomized_at`, `treatment_category`, `ADT_prior`, `mh_prdosu`, `cmgrpid`, `setmcrpc`,
  `cmongo`, `cmstdat_int`, `mh_cmendat_int`, `prohibited_therapy`,
  `prohibited_after_randomization`, `prohibited_ongoing`, `started_tx`, `exstdat_int`,
  `mhsg_prtrt`, `orch_prior`, and `flag_ADT_setting`
29. Affirmation: `RECIST001`
  • `randomized_at`, `treatment_category`, `ADT_prior`, `mh_prdosu`, `cmgrpid`, `setmcrpc`,
  `cmongo`, `cmstdat_int`, `mh_cmendat_int`, `prohibited_therapy`,
  `prohibited_after_randomization`, `prohibited_ongoing`, `started_tx`, `exstdat_int`,
  `mhsg_prtrt`, `orch_prior`, and `flag_doce_mcprc`
30. Affirmation: `RECIST002`
  • `randomized_at`, `treatment_category`, `ADT_prior`, `mh_prdosu`, `cmgrpid`, `setmcrpc`,
  `cmongo`, `cmstdat_int`, `mh_cmendat_int`, `prohibited_therapy`,
  `prohibited_after_randomization`, `prohibited_ongoing`, `started_tx`, `exstdat_int`,
  `mhsg_prtrt`, `orch_prior`, and `flag_doce_dose`
31. Affirmation: `RECIST003`
  • `treatment_category`, `ADT_prior`, `mh_prdosu`, `cmgrpid`, `setmcrpc`, `cmongo`,
  `cmstdat_int`, `mh_cmendat_int`, `prohibited_therapy`, `prohibited_after_randomization`,
  `prohibited_ongoing`, `started_tx`, `exstdat_int`, `mhsg_prtrt`, `orch_prior`, and
  `flag_unmapped`
32. Affirmation: `SRE002`
  • `ADT_prior`, `orch_prior`, and `flag_unmapped`
33. Affirmation: `SS001`
  • `instance_name`, `visit_occurred`, `baseline_present`, `bl_tudat_int`, `fu_tudat_int`,
  `fu_original_source`, `tuloc`, `tuloccat`, `tutypnam`, `tumstate_trtestcd`, `image`,
  `baseline_match`, `flag_source`, `flag_tuloc_bl`, `flag_tutypnam_bl`, `flag_tudat_int_bl`,
  `tumor_name`, `img_tuyn`, …, `flag_lesion`, and `flag_dummy_timepoint`
34. Affirmation: `SS002`
  • `instance_name`, `visit_occurred`, `baseline_present`, `bl_tudat_int`, `fu_tudat_int`,
  `fu_original_source`, `tuloc`, `tuloccat`, `tutypnam`, `tumstate_trtestcd`, `image`,
  `baseline_match`, `flag_source`, `flag_tuloc_bl`, `flag_tutypnam_bl`, `flag_tudat_int_bl`,
  `tumor_name`, `img_tuyn`, …, `flag_lesion`, and `flag_dummy_lesion`
35. Affirmation: `VS001`
  • `ssfu_yn`, `dsdat_death_int`, `fup_cutoff_date`, and `flag_ss`
36. Affirmation: `VS002`
  • `in_vitals`, `instance_name`, `vsdat_int`, `weight_vsorres_std`, `weight_vsorres_std_un`,
  `weight_baseline`, `weight_pct_change`, `ae_expected`, `in_ae`, `aemodify`, and
  `flag_match_ae_detail`

ℹ Please add the missing columns to the current affirm session before attempting to update the
  current report.

Error in `affirm_report_excel()`:
ℹ affirm Excel Report was not updated.
Run `rlang::last_trace()` to see where the error occurred.
> 
```
>[!IMPORTANT]
>The current issue is that the workflow adds "placeholders" of newly created affirmations into the previous reports to facilitate proper joining for the affirmations that _are_ present:

https://github.com/pcctc/affirm/blob/40c3c3b7fc86812882ec3f59341ed3895c9074cb/R/utils-report.R#L374-L388

And this happens before the mismatch check. This may be a matter of making separate lists just for this check, that only compare affirmations that are initially present in both the current and previous reports, or maybe reorganizing the structure of the workflow. Not sure yet.

**Comment by Meghansaha** [comment]
@shannonpileggi, the version of `affirm_report_excel()` in the [most recent commit](https://github.com/pcctc/affirm/pull/44/commits/05675e16ebda51463db3b6acf29c02f65c537e45) now works in production (RE: Incorporating previous comments - https://github.com/pcctc/affirm/issues/41)

> [!IMPORTANT]
> But GH actions are down, so I can't complete these checks for the last commit in this PR:
>
> image

I wanted to discuss this with you anyway. This turned into a messy PR and am thinking I should maybe open a new one. Let's discuss.

**Comment by Meghansaha** [comment]
@shannonpileggi,  This is technically set for review but I'd like to just talk to you about what's going on here before you review... I also figured that this won't be looked at anytime soon.

> [!NOTE]
> This PR modifies the `affirm_report_excel()` function by adding a `previous_file` argument. This modification allows for the inclusion of `comment`, `status`, and `assigned_to` values from a previous affirm report into the creation of a new affirm report.
>
> I believe this modification is substantial enough to bump the version of `{affirm}` to v.0.2.1. The [DESCRIPTION](https://github.com/pcctc/affirm/pull/44/files#diff-9cc358405149db607ff830a16f0b4b21f7366e3c99ec00d52800acebe21b231c) and [NEWS.md](https://github.com/pcctc/affirm/pull/44/files#diff-51920e95310ebfbc1ae31709f3b95f89afffbf4f1a6e38e8b2b406e2fb6197ea) files have been changed to reflect the version bump.

> [!IMPORTANT]
> The bulk of the work is in the new internal `.update_sheets()` function. This function incorporates data from a previous report and joins the older data into the newly created affirmation summary data frame. `.update_sheets()` makes a "modified" version of the "new" summary table that is then processed as usual within the original `affirm_report_excel()` workflow.
> 
> This update has been tested on actual production data/worksheets and can be reproduced by executing the run validation script as explained in the **c16-174 Repository (PR # 426)**

## Major Changes (Lines and Diffs)

<details>
<summary><h2><code>affirm_report.R</code></h2></summary>

### [Diff Here](https://github.com/pcctc/affirm/pull/44/files#diff-5719f08325c1cdfa249de7a7a384475c88fcfed622b8331adf89d0f598c75196)
   -  [Documentation Updates (Lines 12-42)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/R/affirm_report.R#L12-L42)
   -  [`previous_file` Check and Toggle (Lines 86-107)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/R/affirm_report.R#L86-L107)
   -  [`previous_file` Workflow Initiation (Lines 129-141)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/R/affirm_report.R#L129-L141)
   -  [Updates to `status` and `comment` Workflow (Lines 144-185)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/R/affirm_report.R#L144-L185)
</details>

<details>
<summary><h2><code>utils_report.R</code></h2></summary>

### [Diff Here](https://github.com/pcctc/affirm/pull/44/files#diff-da9754c09157418ab14ef5ee04ffe81217038345c1b5f22e811656c5467ad9be)
   -  [`.add_affirmation_sheet` Updates (Lines 195-226)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/R/utils-report.R#L195-L226)
   -  [`.update_sheets()` (Lines 288-778)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/R/utils-report.R#L288-L778) - Major internal function that contains most of the workflow for updating the report.
</details>

<details>
<summary><h2><code>test-affirm_report.R</code></h2></summary>

### [Diff Here](https://github.com/pcctc/affirm/pull/44/files#diff-4af750d0d8888389e14f0a0623169b10fd46649ed78a246e716321c1e8b57b8a)
   -  [Testing Updates and Additions (Lines 378-1350)](https://github.com/pcctc/affirm/blob/18ffb1d2c457de8bf0b825f41544e7b88ed51f76/tests/testthat/test-affirm_report.R#L378-L1350)
</details>

File: R/affirm_report.R [hunk 1]
```diff
@@ -9,10 +9,37 @@
 #' in curly brackets is replaced with the item value (see glue::glue). Item names
 #' accepted include: `id`, `label`, `priority`, `data_frames`, `columns`, `error_n`, `total_n`.
 #' Defaults to `"{data_frames}{id}"`.
+#' @param previous_file A string of the file path to the previous affirmation Excel workbook that needs to be updated
```
Comment (shannonpileggi) [inline R/affirm_report.R]:
```suggestion
#' @param previous_file A string of the file path to the previous affirmation Excel workbook containing `assigned_to`, `status`, and `comment` fields that need to be carried forward to this report
```

File: R/utils-report.R [hunk 2]
```diff
@@ -265,3 +284,495 @@
...
+  lst_prev_affirmation_dfs <- lst_prev_affirmation_dfs[sort(names(lst_prev_affirmation_dfs))]
+
+  #============================================================================#
+  # Mismatched Affirmation Columns Check----------------------------------------
+  #============================================================================#
+  # Create an initial list of new affirmation columns#
+  lst_new_affirmation_cols <-
+    lapply(
+      lst_init_new_affirmation_dfs,
+      function(x) x |> names()
```
Comment (shannonpileggi) [inline R/utils-report.R]:
```suggestion
    lapply(
      lst_init_new_affirmation_dfs,
      names
```

this works equivalently

``` r
lst_tst <- tibble::lst(
  faithful,
  ChickWeight
)


lapply(
  lst_tst,
  function(x) x |> names()
)
#> $faithful
#> [1] "eruptions" "waiting"  
#> 
#> $ChickWeight
#> [1] "weight" "Time"   "Chick"  "Diet"

lapply(
  lst_tst,
  names
)
#> $faithful
#> [1] "eruptions" "waiting"  
#> 
#> $ChickWeight
#> [1] "weight" "Time"   "Chick"  "Diet"
```

<sup>Created on 2025-06-24 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Review by shannonpileggi — APPROVED** [review]
This looks great! Well done figuring this all out, especially with all the edge cases.

My two comments are not super meaningful, just some quick things I noticed.

I didn't truly dig into the code too much, but more focused on tests and outputs and it looks like you covered everything I can think of so far.

The only other thing I'll add is have you watched [Jenny Bryan's code smells and feels talk](https://www.youtube.com/watch?v=7oyiPBjLAWY)? A line in there that stuck with me is "there is no else, there is only if." I'm not saying to refactor this, but just wanted to point it out as a pattern I noticed.

**Comment by Meghansaha** [comment]
Hey @shannonpileggi, The test outputs for the work on this branch have been reviewed and approved. I've also re-ran tests and checks and confirm that everything looks good on my end locally:

Screenshot 2025-07-02 130410

Screenshot 2025-07-02 130848

I feel this is good to be rolled in!
